dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestCriteria.cpp

```diff
@@ -2436,4 +2678,157 @@ TEST(DataInterfaceIIDMTest, testGeneratorCriteriaDataIIDMSum) {
+    criteriaParamsWithVoltageLevel->setScope(vlTestConfig.vlTestScope);
+    criteriaParamsWithVoltageLevel->setPMax(40);
+    criteriaParamsWithVoltageLevel->addVoltageLevel(criteriaVoltageLevel1);
+    criteriaParamsWithVoltageLevel->addVoltageLevel(criteriaVoltageLevel2);
+
+    std::vector<double> busVarray;
+    if (vlTestConfig.filtering) {
+      busVarray = {225, 70};
+    } else {
+      busVarray = {225, 225};
+    }
+
+    // Test on LoadCriteria
+    boost::shared_ptr<powsybl::iidm::Network> loadNetwork = createBusBreakerNetworkWithLoads2(busVarray.at(0),
+                                                                                              busVarray.at(1),
+                                                                                              225,
+                                                                                              190,
+                                                                                              190);
+    boost::shared_ptr<DataInterface> dataWithLoads = createDataItfFromNetworkCriteria(loadNetwork);
+    exportStates(dataWithLoads);
+    const std::vector<shared_ptr<VoltageLevelInterface> > loadVoltageLevels = dataWithLoads->getNetwork()->getVoltageLevels();
+    LoadCriteria loadCriteriaWithoutVoltageLevel(criteriaParamsWithoutVoltageLevel);
+    LoadCriteria loadCriteriaWithVoltageLevel(criteriaParamsWithVoltageLevel);
+    for (size_t vlIdx = 0; vlIdx < loadVoltageLevels.size(); ++vlIdx) {
+      const std::vector<boost::shared_ptr<LoadInterface> > loads = loadVoltageLevels[vlIdx]->getLoads();
+      for (size_t loadIdx = 0; loadIdx < loads.size(); ++loadIdx) {
+        loadCriteriaWithoutVoltageLevel.addLoad(loads[loadIdx]);
+        loadCriteriaWithVoltageLevel.addLoad(loads[loadIdx]);
+      }
+    }
+
+    loadCriteriaWithoutVoltageLevel.checkCriteria(0, vlTestConfig.isFinalStep);
+    const std::vector<std::pair<double, std::string> > loadFailingCriteriaWithoutVL = loadCriteriaWithoutVoltageLevel.getFailingCriteria();
+    ASSERT_EQ(loadFailingCriteriaWithoutVL.size(), vlTestConfig.failingCriteriaSizeWithoutVL);
+    checkNoLoadIsTestedSeveralTimes(loadFailingCriteriaWithoutVL);
+
+    loadCriteriaWithVoltageLevel.checkCriteria(0, vlTestConfig.isFinalStep);
+    const std::vector<std::pair<double, std::string> > loadFailingCriteriaWithVL = loadCriteriaWithVoltageLevel.getFailingCriteria();
+    ASSERT_EQ(loadFailingCriteriaWithVL.size(), vlTestConfig.failingCriteriaSizeWithVL);
+    checkNoLoadIsTestedSeveralTimes(loadFailingCriteriaWithVL);
+
+    if (vlTestConfig.filtering) {
+      if (vlTestConfig.vlTestType == criteria::CriteriaParams::LOCAL_VALUE) {
+        ASSERT_GT(loadFailingCriteriaWithoutVL.size(), loadFailingCriteriaWithVL.size());
+      } else if (vlTestConfig.vlTestType == criteria::CriteriaParams::SUM) {
+        ASSERT_EQ(loadFailingCriteriaWithoutVL.size(), loadFailingCriteriaWithVL.size());
```

Comment:
The size should be 1 right?

Reply:
When the type is SUM, failingCriteria_ size should be 1 with or without VL. It's tested above with these lines : 
```
ASSERT_EQ(loadFailingCriteriaWithoutVL.size(), vlTestConfig.failingCriteriaSizeWithoutVL);
ASSERT_EQ(loadFailingCriteriaWithVL.size(), vlTestConfig.failingCriteriaSizeWithVL);
```

But `loadFailingCriteriaWithoutVL` and `loadFailingCriteriaWithVL` are also supposed to be the same size, that's why I put a test here. Maybe it's overkill ?

Reply:
Then let's explicitely test at some point that those sizes are 1